ee_tests/src/specs/page_objects/main_dashboard.page.ts

```diff
@@ -6,15 +6,165 @@
+    // tslint:enable:max-line-length
+
+    /* Profile page */
+    updateProfileButton = element(by.xpath('.//button[contains (text(), \'Update Profile\')]'));
+    updateTenantButton = element(by.xpath('.//button[contains (text(), \'Update Tenant\')]'));
+
+    /* Profile drop down selection */
+    profile = $('#header_loggedinProfile');
+
+    /* User name or space name in Left Navigation Bar */
+    nameUnderLeftNavigationBar(nameString: string): ElementFinder {
+      let xpath = `.//*[contains(@class, 'navbar-left')]//*[contains(text(), '${nameString}')]`;
+      return element(by.xpath(xpath));
+   }
+
+    // tslint:disable:max-line-length
+    /* Recent space by name */
+    // Example:  .//*[contains (@class, ('recent-items-text-dropdown'))]/span[contains (text(),'testspace')]
+    recentSpaceByName (spaceName: string): ElementFinder {
+      let xpathString = './/*[contains (@class, (\'recent-items-text-dropdown\'))]/span[contains (text(),\'' + spaceName + '\')]';
+      return element(by.xpath(xpathString));
+    }
+    // tslint:enable:max-line-length
+
   async validate() {
     await browser.wait(until.presenceOf(this.appTag));
   }
 
-}
+  /* Helper function to create a new OSIO space */
+  async createNewSpace (targetUrl: string, username: string, spaceName: string) {
+    this.clickElement(this.headerDropDownToggle, 'headerDropDownToggle');
+    browser.sleep(support.WAIT);
+    this.clickElement(this.createSpaceUnderLeftNavigationBar, 'createSpaceUnderLeftNavigationBar');
+    this.sendKeysElement (this.newSpaceName, spaceName);
+    this.sendKeysElement (this.devProcessPulldown, 'Scenario Driven Planning');
+    this.clickElement(this.createSpaceButton, 'createSpaceButton');
+    this.clickElement(this.noThanksButton, 'noThanksButton');
+    browser.getCurrentUrl().then(function (text) {
+        console.log ('EE test - new space URL = ' + text);
+     });
+
+    let urlText = support.joinURIPath(targetUrl, username, spaceName);
+    console.log('Lets wait for the URL to contain the space URL: ' + urlText);
 
+    // tslint:disable:max-line-length
+    browser.wait(until.urlContains(urlText), support.LONG_WAIT, 'Failed waiting to move to the space page with URL: ' + urlText + ' Did create space not work?').then(function () {
+    // tslint:enable:max-line-length
+      browser.getCurrentUrl().then(function (text) {
+         console.log ('The browser was at URL: ' + text);
+      });
+    });
+    browser.wait(until.urlIs(urlText), 10000);
+    expect(browser.getCurrentUrl()).toEqual(urlText);
+    return new SpaceDashboardPage();
+  }
+
+  /* Helper function to click on a UI element */
+  async clickElement (theElement: ElementFinder, elementString: string) {
+    await browser.wait(until.presenceOf(theElement));
+    await browser.wait(until.elementToBeClickable(theElement));
+    theElement.click();
```

Comment:
I think we should await on click() which IIUC will wait for any activity/ http-requests in browser to be over